### PR TITLE
feat(antora): use lunr generator in github action for building docume… …ntation search index

### PR DESCRIPTION
**Pull request by fsismondi**

### CODE REVIEW

**Comment by mcanevet** [comment]
@fsismondi could you please like the diff with upstream? Forking the action means that we'll have to maintain a fork, which we usually try to avoid.

**Comment by fsismondi** [comment]
> @fsismondi could you please like the diff with upstream? Forking the action means that we'll have to maintain a fork, which we usually try to avoid.

The diff with upstream action is the following:

https://github.com/kameshsampath/antora-site-action/compare/master...fsismondi:master

using `lunr` (for the seach index) requires installing a js package, which is not provided by default by `antora`.

a similar gh action to the one I have is this:
https://github.com/ghost-420/antora-lunr-site-action

I wonder do we trust any user for the gh actions ?

**Comment by mcanevet** [comment]
I don't know why @acampergue-camptocamp choose that one a few months back...
Any comment on that @acampergue-camptocamp ?

**Comment by fsismondi** [comment]
I [pinged him](https://github.com/kameshsampath/antora-site-action/issues/4) to see if we can bring changes to the upstream project

**Comment by fsismondi** [comment]
> I don't know why @acampergue-camptocamp choose that one a few months back...
> Any comment on that @acampergue-camptocamp ?

 kameshsampath project looks good to me. Maybe for enhancing security we could reference directly SHA-1 for external actions

**Review by acampergue-camptocamp — COMMENTED** [review]
@mcanevet no specific reason, I took an example that worked

**Comment by mcanevet** [comment]
Honestly I'm not qualified to either approve or reject this PR. I let that to @acampergue-camptocamp and/or @raphink .

**Comment by mcanevet** [comment]
@fsismondi is this change still relevant?

**Comment by fsismondi** [comment]
@mcanevet I think it is. No news from author I guess we can use the forked action?

**Comment by mcanevet** [comment]
OK for me, @raphink any opinion?

File: .github/workflows/website.yml [hunk 1]
```diff
@@ -37,9 +37,11 @@ jobs:
         with:
           fetch-depth: 0   # Fetch all history for .gitinfo and .lastmod
       - name: "Generate site using antora site action"
-        uses: kameshsampath/antora-site-action@master
+        uses: fsismondi/antora-site-action@master
```
Comment (mcanevet) [inline .github/workflows/website.yml:40]: You need at least to fork into Camptocamp

**Review by mcanevet — CHANGES_REQUESTED** [review]

**Comment by mcanevet** [comment]
@fsismondi did you take a look at https://github.com/ghost-420/antora-lunr-site-action?

**Comment by fsismondi** [comment]
> @fsismondi did you take a look at https://github.com/ghost-420/antora-lunr-site-action?

yep, back at the time (see comment May 5th) 

I'll test again, if that action seems enough I'll just proceed with that

File: .github/workflows/website.yml (hunk 1 above)
Comment (raphink) [reply .github/workflows/website.yml:40]: Has there been a PR sent upstream to integrate your changes?

Comment (fsismondi) [reply .github/workflows/website.yml:40]: https://github.com/kameshsampath/antora-site-action/pull/6 
\o/

Ill proceed with the updates on our side

**Comment by fsismondi** [comment]
Contrib merged upstream https://github.com/kameshsampath/antora-site-action/pull/6 Ok for merge?

**Review by mcanevet — APPROVED** [review]

**Pull request merged**